### ISSUE TITLE
关于EmbeddingIndex函数的疑问

### ISSUE THREAD

**Issue by formygreatking** [shenweichen/DeepMatch#20]
**Describe the question(问题描述)**
class EmbeddingIndex(Layer):

    def __init__(self, index,**kwargs):
        self.index =index
        super(EmbeddingIndex, self).__init__(**kwargs)

    def build(self, input_shape):

        super(EmbeddingIndex, self).build(
            input_shape)  # Be sure to call this somewhere!
    def call(self, x, **kwargs):
       return tf.constant(self.index)
    def get_config(self, ):
        config = {'index': self.index, }
        base_config = super(EmbeddingIndex, self).get_config()
        return dict(list(base_config.items()) + list(config.items()))

请问一层返回的是embedding index吗？从call函数里的逻辑看，返回的是整个index list

**Comment by ucasiggcas** [comment]
同样我觉得NoMask似乎没啥用。
```
class NoMask(tf.keras.layers.Layer):
    def __init__(self, **kwargs):
        super(NoMask, self).__init__(**kwargs)

    def build(self, input_shape):
        # Be sure to call this somewhere!
        super(NoMask, self).build(input_shape)

    def call(self, x, mask=None, **kwargs):
        return x

    def compute_mask(self, inputs, mask):
        return None
```

**Comment by skriser** [comment]
是的，看不出来EmbeddingIndex有什么具体作用。难道是为了以后用吗

**Comment by bbruceyuan** [comment]
作用就是把 item对应的 index 作为 label。这是为了配合 [SmapledSoftmax](https://www.tensorflow.org/api_docs/python/tf/nn/sampled_softmax_loss)，其中有一个参数是 Label, SampledSoftmax参数中的 Label 不是 one-hot, 而是 item 的 index
